rdmd.d

```diff
@@ -713,9 +725,104 @@ private string[string] getDependencies(string rootModule, string workDir,
         exit(depsExitCode);
     }
 
+    if (!hasDependencies)
+        scope(exit) std.file.remove(rootModule);
+
     return dryRun ? null : readDepsFile();
 }
 
+bool needsInspection(string file)
+{
+    foreach (match; file.matchAll(ctRegex!`import\s+(.*);`))
+    {
+        import std.utf : byChar;
+        // selective imports can only be at the end of an ImportList
+        // hence we can just chop of the imports
+        auto selectiveImportStart = match[1].byChar.countUntil(':');
+        if (selectiveImportStart < 0)
+            selectiveImportStart = match[1].length;
+
+        foreach (importText; match[1][0..selectiveImportStart].splitter(","))
+        {
+            string mod;
+
+            // renamed imports
+            if (auto renamedImport = importText.matchFirst(ctRegex!`[^=]*=\s*(.*)`))
+                mod = renamedImport[1];
+            else
+                mod = importText.strip;
+
+            if (!(mod.startsWith("std") || mod.startsWith("core")))
+            {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+// true positive imports from the standard library or runtime
+unittest
+{
+    assert(!"import std.traits;".needsInspection);
+    assert(!"import core.runtime;".needsInspection);
+    assert(!"public import   std.traits;".needsInspection);
+
+    // multiple imports
+    assert(!"import std.traits, std.math;".needsInspection);
+    assert(!"import std.traits ,std.math, core.runtime;".needsInspection);
+
+    // selective imports
+    assert(!"import std.math: exp;".needsInspection);
+    assert(!"import std.math : cos, exp, sin;".needsInspection);
+
+    // selective imports can be the last part of an ImportList
+    assert(!"import std.traits, std.math : exp;".needsInspection);
+    assert(!"import std.traits, std.math : cos, exp, sin;".needsInspection);
+
+    // renamed imports
+    assert(!"import io = std.stdio;".needsInspection);
+    assert(!"import io = std.stdio, math = std.math;".needsInspection);
+
+    // selective renamed imports
+    assert(!"import stdio = std.stdio: writeln, dump = write;".needsInspection);
+    assert(!"import traits = std.traits: isFloatingPoint;".needsInspection);
+    assert(!"import math = std.math, stdio = std.stdio: writeln, dump = write;".needsInspection);
+}
+
+// false positive imports from the third-party libraries
```

Comment:
"false positive" implies a wrong result. So do the unit tests check against false positives/negatives, or do they ensure that the tests correctly return true positives/negatives?